src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -381,7 +440,12 @@ private TotalHits getTotalHits(int trackTotalHitsUpTo, final List<?> topDocs, fi
         return new TotalHits(maxTotalHits, relation);
     }
 
-    private TopDocs getNewTopFieldDocs(final TotalHits totalHits, final List<TopFieldDocs> topFieldDocs, final SortField sortFields[]) {
+    private TopDocs getNewTopFieldDocs(
```

Comment:
Can we create a issue to make a generic class which will decide which method to call?
1. getTopDocs
2. getTopFieldDocs
3. getCollapseTopFieldDocs

In that issue we can seperate this coillapse code to a dedicated method and the generic class would decide on the basis of which functionality to use.

cc: @martin-gaievski @owaiskazi19